internal/collector/sync_quota_to_backend.go

```diff
@@ -173,50 +157,30 @@ func (c *Collector) performQuotaSync(ctx context.Context, srv db.ProjectService,
 			return nil
 		}
 
-		var targetQuota uint64
-		if resInfo.Topology == liquid.AZSeparatedTopology {
-			// for AZSeparatedTopology, project_resources.quota is effectively empty (always set to zero)
-			// and `targetQuota` needs to be computed by summing over project_az_resources.quota
-			targetQuota = 0
-			err = sqlext.ForeachRow(c.DB, azQuotaSyncSelectQuery, []any{resourceID, project.ID}, func(rows *sql.Rows) error {
-				var (
-					availabilityZone liquid.AvailabilityZone
-					currentAZQuota   *int64
-					targetAZQuota    uint64
-				)
-				err := rows.Scan(&availabilityZone, &currentAZQuota, &targetAZQuota)
-				if err != nil {
-					return err
-				}
-				// defense in depth: configured backend_quota for AZ any or unknown are not valid for the azSeparatedQuota topology.
-				if (availabilityZone == liquid.AvailabilityZoneAny || availabilityZone == liquid.AvailabilityZoneUnknown) && currentAZQuota != nil {
-					return fmt.Errorf("detected invalid AZ: %s for resource: %s with topology: %s has backend_quota: %v", availabilityZone, resourceName, resInfo.Topology, currentAZQuota)
-				}
-				azSeparatedResourceIDs = append(azSeparatedResourceIDs, resourceID)
-				if targetAZQuotasInDB[resourceName] == nil {
-					targetAZQuotasInDB[resourceName] = make(map[liquid.AvailabilityZone]liquid.AZResourceQuotaRequest)
-				}
-				targetAZQuotasInDB[resourceName][availabilityZone] = liquid.AZResourceQuotaRequest{Quota: targetAZQuota}
-				targetQuota += targetAZQuota
-				if currentAZQuota == nil || *currentAZQuota < 0 || uint64(*currentAZQuota) != targetAZQuota {
-					azSeparatedNeedsApply = true
-				}
-				return nil
-			})
-			if err != nil {
-				return err
-			}
+		targetQuota, targetQuotaExists := targetQuotaPtr.Unpack()
+		if !targetQuotaExists {
+			return fmt.Errorf("found unexpected NULL value in project_az_resources.quota for id = %d", projectAZResourceID)
+		}
+		currentQuota, currentQuotaExists := currentQuotaPtr.Unpack()
+		// defense in depth: configured backend_quota for AZ any or unknown are not valid for the azSeparatedQuota topology.
+		if resInfo.Topology == liquid.AZSeparatedTopology && (availabilityZone == liquid.AvailabilityZoneAny || availabilityZone == liquid.AvailabilityZoneUnknown) && !currentQuotaExists {
+			return fmt.Errorf("detected invalid AZ: %s for resource: %s with topology: %s has backend_quota: %v", availabilityZone, resourceName, resInfo.Topology, currentQuota)
```

Comment:
Same here. `resourceName` alone might be ambiguous (e.g. `capacity`), so we should include the service type. Also, I'm not a huge fan of the colon-overusing message style. Also also, `currentQuota` is not well-defined in this branch because we know that `currentQuotaExists == false`.
```suggestion
			return fmt.Errorf("detected invalid AZ %q for resource %s/%s with topology %s (backend quota was nil)", availabilityZone, serviceType, resourceName, resInfo.Topology)
```